spec.bs

```diff
@@ -1676,14 +1699,15 @@ and a [=real time reporting contributions map=] |realTimeContributionsMap|:
   1. [=list/For each=] |component| in |auctionConfig|'s [=auction config/component auctions=],
     [=parallel queue/enqueue steps|enqueue the following steps=] to |queue|:
     1. If |component|'s [=auction config/server response=] is not null:
-      1. Let |compWinnerInfo| be the result of running [=parse and validate server response=] with |component|,
-         |auctionConfig|, |global|, |bidIgs|, and |bidDebugReportInfoList|.
+      1. Let |compWinnerInfo| be the result of running [=parse and validate server response=] with
+         |component|, |auctionConfig|, |global|, |bidIgs|, and |bidDebugReportInfoList|.
     1. Otherwise:
       1. Let |compWinnerInfo| be the result of running [=generate and score bids=] with |component|,
         |auctionConfig|, |global|, |bidIgs|, |bidDebugReportInfoList|, and |realTimeContributionsMap|.
-    1. If |compWinnerInfo| is failure, abort these steps.
-    1. If [=recursively wait until configuration input promises resolve=] given |auctionConfig| returns
-      failure, abort these steps.
+    1. If |compWinnerInfo| is failure, or [=recursively wait until configuration input promises resolve=]

```

Comment:
@morlovich After reading it more, I think we need to decrement |pendingComponentAuctions| in these cases, otherwise the step below that waits for |pendingComponentAuctions| to be 0 will keep waiting?

Reply:
Good catch. I think we may abort less then the implementation (just a component and not the whole thing) when this happens, too?

Reply:
hmm yes, but if we check [=recursively wait until configuration input promises resolve=] first before "|compWinnerInfo| is failure" here, any component promise failure will fail all component auctions (not top level though, but top level would have no winner from component auctions)?
I think we've talked about this that the implementation of a component promise failure failing the whole auction is not perfect. Is there a plan to change that? I guess no, if that's the case, we'd be better to make the spec match the impl.

Reply:
It seems component auction server response promise only fail a single component auction, not the whole auction (if I read https://crsrc.org/c/content/browser/interest_group/interest_group_auction.cc;drc=f522344e45882da4c7f7cb1b3a0a7bd747d654bb;l=5760 correctly, although not fully sure).

Reply:
ah interesting. I think it would be better to make it consistent for this single promise? WDYT? @brusshamilton FYI as well

Reply:
Yes, it probably should just fail the component auction.

Reply:
ok, (1) I changed it to check [=recursively wait until configuration input promises resolve=] first so that any (non server response) promise failure could fail the whole auction which meets our implementation (the impl's behaviour is not ideal but it's not easy to change and there's no plan to change it yet).
(2) For server response promise, I kept the behaviour of only fail the component auction no matter what was the failure (promise reject, resovled promise parse failure, etc,). It's different from our impl since promise reject fails the whole auction, but we do have a plan to change the impl and it feels doable. I can change the spec if the plan changes.